### ISSUE TITLE
Feature request - View dataframes from debugging in Data Wrangler

### ISSUE THREAD

**Issue by idling-mind** [microsoft/vscode-data-wrangler#49]
hi,
  In vscode, it would be nice to be able to open a dataframe from the debug window in Data Wrangler. As of now, it only gives an option to open a dataframe in data viewer.

**Comment by idling-mind** [comment]
hi,
When i'm debugging a python file, I can inspect the variables in the debug variable inspector. If one of the variable is a dataframe, 
I could open that in a data viewer and explore it in a separate window. 
![image](https://github.com/microsoft/vscode-data-wrangler/assets/13498682/44e5d157-ace9-41f2-972d-91785a301da2)

The data viewer is quite nice; but probably not as cool as data wrangler.  Hope this explains the idea.

**Comment by pwang347** [comment]
Hi @idling-mind, that makes sense - thanks for the clarification! I've created an issue here https://github.com/microsoft/vscode-jupyter/issues/13626 as we will need an API from the Jupyter team.

**Comment by chilipepperpizza** [comment]
I saw a note in VS Code today stating that the Data Viewer will be depreciated/removed soon and suggested using this extension instead. Without Data Viewer (and without this feature request) will there be no way to view the contents of a dataframe when debugging a .py file?

**Comment by pwang347** [comment]
> I saw a note in VS Code today stating that the Data Viewer will be depreciated/removed soon and suggested using this extension instead. Without Data Viewer (and without this feature request) will there be no way to view the contents of a dataframe when debugging a .py file?

Hi @chilipepperpizza! Just wanted to confirm that this is a scenario that we plan to support soon - the feature is merged and released on our side. The corresponding changes are also merged in Jupyter but yet to be released, so I'll let you know once it's available in pre-release.

For some more context, you can check out https://github.com/microsoft/vscode-data-wrangler/issues/131

**Comment by chilipepperpizza** [comment]
That's great news, thank you!

**Comment by idling-mind** [comment]
hi @pwang347,
  I have installed the pre-release version of data-wrangler. However, I have not yet figured out how to open a debug variable (which is a dataframe) in data-wrangler yet. Can you please explain or show a screenshot?

**Comment by juanbretti** [comment]
> hi @pwang347,
> I have installed the pre-release version of data-wrangler. However, I have not yet figured out how to open a debug variable (which is a dataframe) in data-wrangler yet. Can you please explain or show a screenshot?

Do you have installed [Visual Studio Code Insiders](https://code.visualstudio.com/insiders/)?, as mentioned at @pwang347  post.

**Comment by idling-mind** [comment]
@juanbretti , thanks for pointing that out. I missed that. Will try again after installing vscode-insiders.

**Comment by pwang347** [comment]
@idling-mind Yes, VS Code Insiders will be needed until around next week when VS Code updates!

To clarify additionally, you should be able to use the same button in your original post here:
![image](https://github.com/microsoft/vscode-data-wrangler/assets/8560030/8415d530-5c40-4a8f-8c76-74dd8eae4bbe)

Clicking it will now route to Data Wrangler's viewer when both of the latest pre-release extensions are installed in VS Code Insiders.

**Comment by DaveGuenther** [comment]
Hi Folks,

I'm currently running [coder-server 4.89.1 (code 1.89.1)](https://github.com/coder/code-server/releases/tag/v4.89.1), and connecting to it via web browser.  Will this version be able to work with the new Data Wrangler to get the "View Value in Data Viewer" functionality or will I need to update code-server to a newer version for the web functionality?

Thanks!

**Comment by nabilkhalid56** [comment]
I am looking for a data viewer that can also handle complex variables. Jupyter Notebook shows complex variables as [object], therefore, I just tried data wrangler pre-release in vscode Insider. Although I can see all the variables and their values, but the viewer somehow shows the real numbers as complex numbers. Could this be fixed?

![image](https://github.com/microsoft/vscode-data-wrangler/assets/35198306/166a3d42-df6b-4724-a550-e3d62aec63c3)

**Comment by pwang347** [comment]
Hi @nabilkhalid56, thanks for letting me know! If possible, could you please copy the details you shared above into a new issue and also include a small code snippet on how to recreate the issue?

That would help expedite the process of us getting a fix out. Thanks!

**Comment by dkarelov** [comment]
Is it still not available in the release version of the VS Code?
Routing is still to the old data viewer, even if pre-release version of Data Wrangler is used.

**Comment by pwang347** [comment]
Hi @dkarelov, yes it should be already available in stable.

Could you please double check to see if you have the following configuration:

**VS Code Stable**: 1.91.0 or higher
**Data Wrangler Extension**: 1.4.2 or higher
**Jupyter Extension**: 2024.6.0 or higher

If this is true, but you are still being routed to the old viewer then I would recommend [opening an issue on the Jupyter repository](https://github.com/microsoft/vscode-jupyter/issues) as the routing happens from that extension.

For more context please see the comments in this other thread: https://github.com/microsoft/vscode-data-wrangler/issues/131#issuecomment-2214735799

Thanks!

**Comment by UdaykiranEstari** [comment]
Hi @pwang347 I'm not able to see the option view **data in data viewer**. for in memory dataframes while I'm debugging .py files. Is there any work around for this.

**Comment by danhje** [comment]
Same here, I work with .py files, not Notebooks, and would like to use Data Wrangler in debug mode. There's no "view data i data viewer" when I right click a DataFrame variable.

**Comment by pwang347** [comment]
Hi @UdaykiranEstari, @danhje - do you have the [Jupyter extension](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter) installed? The button is currently coming from their side (even if you don't use IPython or Jupyter). We are also looking into potentially having our own launch button in the future, but for now we are using theirs as it used to redirect to the old data viewer. Please let me know if installing that extension makes the button appear for you. Thanks!

**Comment by danhje** [comment]
It works for me now.

I did have the Nypyter extension installed, both locally and in the WSL distro (I run VSCode on Windows against WSL), and I restarted the IDE, but it still didn't work, until it suddenly did. The only explanation I can think of is that the VSCode remote server didn't restart the first times I restarted my IDE.
